Added a request that provides sessions for given user.

PetrutaRus opened the pull request.

PetrutaRus committed:
Added a request that provides sessions for given user.

Florin-Catalin commented on src/main/java/com/bootcamp/demo/UserController.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+
+}
```
the controller depends on how the service will look like ( where all the interesting things will take place ). So this will be refactor again after we have the implementation, and it will just return the result of the service call

PetrutaRus committed:
Removed unused code.

PetrutaRus committed:
Added service for sessions.

Florin-Catalin reviewed (approved):
seems fine to me

denisadragota commented:
Avem nevoie si de un SessionService ? Ma gandeam ca putem integra metoda aceea de getAllSessions(user) direct in LoginService, din moment ce oricum are acces la ambele Repository-uri.

PetrutaRus commented:
> Avem nevoie si de un SessionService ? Ma gandeam ca putem integra metoda aceea de getAllSessions(user) direct in LoginService, din moment ce oricum are acces la ambele Repository-uri.

Pe master nu exista nici un service, nu pot folosi ceva ce nu exista. In plus consider ca login servise  trebuie sa se ocupe doar de log in , nu si de managementul sesiunilor.

denisadragota commented:
In LogIn Service, in metodele de logIn si logOut, este nevoie de creare de sesiune noua, respectiv update al unei sesiuni existente (te poti uita la pull request-ul meu care e de cateva zile). De aceea spuneam ca LogInService foloseste sesiunile in mod direct si s-ar putea ocupa el si de asta. O alta varianta este, daca vrei sa separam Serviciile, sa scot din LogInService crearea si actualizarea sesiunilor, si sa facem acest lucru doar in Controller.

PetrutaRus committed:
Merge branch 'main' into SessionList

PetrutaRus committed:
Merge branch 'main' into SessionList

PetrutaRus committed:
Merge branch 'fixed-build-service-repository-integration' into SessionList

PetrutaRus committed:
Fixed code after merge.

PetrutaRus committed:
Merge branch 'main' into SessionList

PetrutaRus committed:
Merge branch 'UserValidation' of github.com:minerva-mcgonagall-team/java-bootcamp-login into SessionList

PetrutaRus committed:
fixed CORS

PetrutaRus committed:
Merge branch 'main' into SessionList

PetrutaRus committed:
Fixed conflicts and fixed dependency injection

PetrutaRus commented:
I fixed interaction with CORS ,I tested my implementation with mocked data on a local react server .

After a checking up with the main branch I observed a that strange repository factory, I refactored the code to use  a proper dependency injection by adding firebase bean config.

PetrutaRus committed:
Merge branch 'main' into SessionList

# Conflicts:
#	src/main/java/com/bootcamp/demo/FirebaseController.java
#	src/main/java/com/bootcamp/demo/controller/ControllerLogin.java
#	src/main/java/com/bootcamp/demo/model/User.java
#	src/main/java/com/bootcamp/demo/security/SecurityConfig.java
#	src/main/java/com/bootcamp/demo/service/LoginService.java

PetrutaRus committed:
Fixed build
